client/my-sites/plan-features-2023-grid/plan-comparison-grid.tsx

```diff
@@ -373,16 +377,31 @@ export const PlanComparisonGrid: React.FC< PlanComparisonGridProps > = ( {
 	useEffect( () => {
 		let newVisiblePlans = displayedPlansProperties.map( ( { planName } ) => planName );
 
-		if ( isMediumBreakpoint ) {
-			newVisiblePlans = newVisiblePlans.slice( 0, 4 );
+		let visibleLength = newVisiblePlans.length;
+		if ( ! isInSignup ) {
+			visibleLength = isLargestBreakpoint ? 4 : visibleLength;
+			visibleLength = isLargeBreakpoint ? 3 : visibleLength;
 		}
 
-		if ( isSmallBreakpoint || ( isInSignup && isMediumBreakpoint ) ) {
-			newVisiblePlans = newVisiblePlans.slice( 0, 2 );
+		visibleLength = isMediumBreakpoint ? 2 : visibleLength;
+
+		if ( newVisiblePlans.length !== visibleLength ) {
+			const currentPlan = displayedPlansProperties.find( ( { current } ) => current );
+			if ( currentPlan ) {
+				newVisiblePlans.sort( ( a ) => ( a === currentPlan.planName ? -1 : 1 ) );
+			}
+			newVisiblePlans = newVisiblePlans.slice( 0, visibleLength );
```

Comment:
> otherwise, maybe avoiding the reassignments might force us to name an extra variable or two and speak more clearly.

I'm not sure what you mean, but I agree this can be improved. I followed your [idea below](https://github.com/Automattic/wp-calypso/pull/72774#discussion_r1091035805) and simplified by using the already available `currentSitePlanSlug`.

I agree this can be simplified by refactoring `usePricingBreakpoint` (or even replacing it with a CSS solution) but considering the time constraints, it looks more like a post-launch effort.